File: commonLib/src/main/java/com/travel/commonlib/hotel/PaymentResults.java

package com.travel.commonlib.hotel;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class PaymentResults  implements Serializable {

    @SerializedName("SelectedPaymentId")
    @Expose
    public int selectedPaymentId;
    @SerializedName("TransactionId")
    @Expose
    public int transactionId;
    @SerializedName("PaymentTransactionId")
    @Expose
    public int paymentTransactionId;
    @SerializedName("DirectPaymentId")
    @Expose
    public int directPaymentId;
    @SerializedName("PaymentUrl")
    @Expose
    public String paymentUrl;
    @SerializedName("PaymentForm")
    @Expose
    public String paymentForm;
    @SerializedName("IsPaymentSuccess")
    @Expose
    public boolean isPaymentSuccess;
    @SerializedName("Result")
    @Expose
    public Object result;
    @SerializedName("Currency")
    @Expose
    public String currency;
    @SerializedName("Amount")
    @Expose
    public double amount;
    @SerializedName("Message")
    @Expose
    public String message;
    @SerializedName("PaymentId")
    @Expose
    public Object paymentId;
    @SerializedName("TransactionNumber")
    @Expose
    public Object transactionNumber;
    @SerializedName("ReceiptNo")
    @Expose
    public Object receiptNo;
    @SerializedName("TrackId")
    @Expose
    public Object trackId;
    @SerializedName("TransactionTime")
    @Expose
    public Object transactionTime;
    @SerializedName("AuthorizeId")
    @Expose
    public Object authorizeId;
    @SerializedName("Description")
    @Expose
    public Object description;
    @SerializedName("ReferenceNumber")
    @Expose
    public Object referenceNumber;
    @SerializedName("BatchNo")
    @Expose
    public Object batchNo;
    @SerializedName("kfhsessionId")
    @Expose
    public Object kfhsessionId;
    @SerializedName("CrediMaxsessionId")
    @Expose
    public Object crediMaxsessionId;
    @SerializedName("kfhDetails")
    @Expose
    public Object kfhDetails;
    @SerializedName("CrediMaxDetails")
    @Expose
    public Object crediMaxDetails;
    @SerializedName("CcAvenueDetails")
    @Expose
    public Object ccAvenueDetails;
    @SerializedName("kfhString")
    @Expose
    public Object kfhString;
    @SerializedName("CardStartNo")
    @Expose
    public Object cardStartNo;
    @SerializedName("CardEndNo")
    @Expose
    public Object cardEndNo;
    @SerializedName("CardType")
    @Expose
    public Object cardType;
    @SerializedName("CardIssuer")
    @Expose
    public Object cardIssuer;
    @SerializedName("BankName")
    @Expose
    public Object bankName;
    @SerializedName("PayedByName")
    @Expose
    public Object payedByName;
    @SerializedName("PaymentStatus")
    @Expose
    public int paymentStatus;
    @SerializedName("CardBasicInfo")
    @Expose
    public Object cardBasicInfo;
    @SerializedName("IsPayLater")
    @Expose
    public boolean isPayLater;
    @SerializedName("IsPayLaterTicket")
    @Expose
    public boolean isPayLaterTicket;
    @SerializedName("IsDummyTicket")
    @Expose
    public boolean isDummyTicket;
    @SerializedName("PaymentSubSectionInfo")
    @Expose
    public Object paymentSubSectionInfo;
    @SerializedName("IsCommonMultiPayment")
    @Expose
    public boolean isCommonMultiPayment;
    @SerializedName("TransactionTypeId")
    @Expose
    public int transactionTypeId;
    @SerializedName("olpIdAlias")
    @Expose
    public Object olpIdAlias;

    public int getSelectedPaymentId() {
        return selectedPaymentId;
    }

    public void setSelectedPaymentId(int selectedPaymentId) {
        this.selectedPaymentId = selectedPaymentId;
    }

    public int getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(int transactionId) {
        this.transactionId = transactionId;
    }

    public int getPaymentTransactionId() {
        return paymentTransactionId;
    }

    public void setPaymentTransactionId(int paymentTransactionId) {
        this.paymentTransactionId = paymentTransactionId;
    }

    public int getDirectPaymentId() {
        return directPaymentId;
    }

    public void setDirectPaymentId(int directPaymentId) {
        this.directPaymentId = directPaymentId;
    }

    public String getPaymentUrl() {
        return paymentUrl;
    }

    public void setPaymentUrl(String paymentUrl) {
        this.paymentUrl = paymentUrl;
    }

    public String getPaymentForm() {
        return paymentForm;
    }

    public void setPaymentForm(String paymentForm) {
        this.paymentForm = paymentForm;
    }

    public boolean isPaymentSuccess() {
        return isPaymentSuccess;
    }

    public void setPaymentSuccess(boolean paymentSuccess) {
        isPaymentSuccess = paymentSuccess;
    }

    public Object getResult() {
        return result;
    }

    public void setResult(Object result) {
        this.result = result;
    }

    public String getCurrency() {
        return currency;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Object getPaymentId() {
        return paymentId;
    }

    public void setPaymentId(Object paymentId) {
        this.paymentId = paymentId;
    }

    public Object getTransactionNumber() {
        return transactionNumber;
    }

    public void setTransactionNumber(Object transactionNumber) {
        this.transactionNumber = transactionNumber;
    }

    public Object getReceiptNo() {
        return receiptNo;
    }

    public void setReceiptNo(Object receiptNo) {
        this.receiptNo = receiptNo;
    }

    public Object getTrackId() {
        return trackId;
    }

    public void setTrackId(Object trackId) {
        this.trackId = trackId;
    }

    public Object getTransactionTime() {
        return transactionTime;
    }

    public void setTransactionTime(Object transactionTime) {
        this.transactionTime = transactionTime;
    }

    public Object getAuthorizeId() {
        return authorizeId;
    }

    public void setAuthorizeId(Object authorizeId) {
        this.authorizeId = authorizeId;
    }

    public Object getDescription() {
        return description;
    }

    public void setDescription(Object description) {
        this.description = description;
    }

    public Object getReferenceNumber() {
        return referenceNumber;
    }

    public void setReferenceNumber(Object referenceNumber) {
        this.referenceNumber = referenceNumber;
    }

    public Object getBatchNo() {
        return batchNo;
    }

    public void setBatchNo(Object batchNo) {
        this.batchNo = batchNo;
    }

    public Object getKfhsessionId() {
        return kfhsessionId;
    }

    public void setKfhsessionId(Object kfhsessionId) {
        this.kfhsessionId = kfhsessionId;
    }

    public Object getCrediMaxsessionId() {
        return crediMaxsessionId;
    }

    public void setCrediMaxsessionId(Object crediMaxsessionId) {
        this.crediMaxsessionId = crediMaxsessionId;
    }

    public Object getKfhDetails() {
        return kfhDetails;
    }

    public void setKfhDetails(Object kfhDetails) {
        this.kfhDetails = kfhDetails;
    }

    public Object getCrediMaxDetails() {
        return crediMaxDetails;
    }

    public void setCrediMaxDetails(Object crediMaxDetails) {
        this.crediMaxDetails = crediMaxDetails;
    }

    public Object getCcAvenueDetails() {
        return ccAvenueDetails;
    }

    public void setCcAvenueDetails(Object ccAvenueDetails) {
        this.ccAvenueDetails = ccAvenueDetails;
    }

    public Object getKfhString() {
        return kfhString;
    }

    public void setKfhString(Object kfhString) {
        this.kfhString = kfhString;
    }

    public Object getCardStartNo() {
        return cardStartNo;
    }

    public void setCardStartNo(Object cardStartNo) {
        this.cardStartNo = cardStartNo;
    }

    public Object getCardEndNo() {
        return cardEndNo;
    }

    public void setCardEndNo(Object cardEndNo) {
        this.cardEndNo = cardEndNo;
    }

    public Object getCardType() {
        return cardType;
    }

    public void setCardType(Object cardType) {
        this.cardType = cardType;
    }

    public Object getCardIssuer() {
        return cardIssuer;
    }

    public void setCardIssuer(Object cardIssuer) {
        this.cardIssuer = cardIssuer;
    }

    public Object getBankName() {
        return bankName;
    }

    public void setBankName(Object bankName) {
        this.bankName = bankName;
    }

    public Object getPayedByName() {
        return payedByName;
    }

    public void setPayedByName(Object payedByName) {
        this.payedByName = payedByName;
    }

    public int getPaymentStatus() {
        return paymentStatus;
    }

    public void setPaymentStatus(int paymentStatus) {
        this.paymentStatus = paymentStatus;
    }

    public Object getCardBasicInfo() {
        return cardBasicInfo;
    }

    public void setCardBasicInfo(Object cardBasicInfo) {
        this.cardBasicInfo = cardBasicInfo;
    }

    public boolean isPayLater() {
        return isPayLater;
    }

    public void setPayLater(boolean payLater) {
        isPayLater = payLater;
    }

    public boolean isPayLaterTicket() {
        return isPayLaterTicket;
    }

    public void setPayLaterTicket(boolean payLaterTicket) {
        isPayLaterTicket = payLaterTicket;
    }

    public boolean isDummyTicket() {
        return isDummyTicket;
    }

    public void setDummyTicket(boolean dummyTicket) {
        isDummyTicket = dummyTicket;
    }

    public Object getPaymentSubSectionInfo() {
        return paymentSubSectionInfo;
    }

    public void setPaymentSubSectionInfo(Object paymentSubSectionInfo) {
        this.paymentSubSectionInfo = paymentSubSectionInfo;
    }

    public boolean isCommonMultiPayment() {
        return isCommonMultiPayment;
    }

    public void setCommonMultiPayment(boolean commonMultiPayment) {
        isCommonMultiPayment = commonMultiPayment;
    }

    public int getTransactionTypeId() {
        return transactionTypeId;
    }

    public void setTransactionTypeId(int transactionTypeId) {
        this.transactionTypeId = transactionTypeId;
    }

    public Object getOlpIdAlias() {
        return olpIdAlias;
    }

    public void setOlpIdAlias(Object olpIdAlias) {
        this.olpIdAlias = olpIdAlias;
    }
}

File: commonLib/src/main/java/com/travel/commonlib/activity/setpayment/PaymentResults.kt
package com.travel.commonlib.activity.setpayment

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class PaymentResults : Serializable {
    @SerializedName("SelectedPaymentId")
    @Expose
    var selectedPaymentId: Long = 0

    @SerializedName("TransactionId")
    @Expose
    var transactionId: Long = 0

    @SerializedName("PaymentTransactionId")
    @Expose
    var paymentTransactionId: Long = 0

    @SerializedName("DirectPaymentId")
    @Expose
    var directPaymentId: Long = 0

    @SerializedName("PaymentUrl")
    @Expose
    var paymentUrl: String? = null

    @SerializedName("PaymentForm")
    @Expose
    var paymentForm: Any? = null

    @SerializedName("IsPaymentSuccess")
    @Expose
    var isIsPaymentSuccess = false
        private set

    @SerializedName("Result")
    @Expose
    var result: Any? = null

    @SerializedName("Currency")
    @Expose
    var currency: String? = null

    @SerializedName("Amount")
    @Expose
    var amount = 0.0

    @SerializedName("Message")
    @Expose
    var message: Any? = null

    @SerializedName("PaymentId")
    @Expose
    var paymentId: Any? = null

    @SerializedName("TransactionNumber")
    @Expose
    var transactionNumber: Any? = null

    @SerializedName("ReceiptNo")
    @Expose
    var receiptNo: Any? = null

    @SerializedName("TrackId")
    @Expose
    var trackId: Any? = null

    @SerializedName("TransactionTime")
    @Expose
    var transactionTime: Any? = null

    @SerializedName("AuthorizeId")
    @Expose
    var authorizeId: Any? = null

    @SerializedName("Description")
    @Expose
    var description: Any? = null

    @SerializedName("ReferenceNumber")
    @Expose
    var referenceNumber: Any? = null

    @SerializedName("BatchNo")
    @Expose
    var batchNo: Any? = null

    @SerializedName("kfhsessionId")
    @Expose
    var kfhsessionId: Any? = null

    @SerializedName("CrediMaxsessionId")
    @Expose
    var crediMaxsessionId: String? = null

    @SerializedName("kfhDetails")
    @Expose
    var kfhDetails: Any? = null

    @SerializedName("CrediMaxDetails")
    @Expose
    var crediMaxDetails: Any? = null

    @SerializedName("CcAvenueDetails")
    @Expose
    var ccAvenueDetails: Any? = null

    @SerializedName("kfhString")
    @Expose
    var kfhString: Any? = null

    @SerializedName("CardStartNo")
    @Expose
    var cardStartNo: Any? = null

    @SerializedName("CardEndNo")
    @Expose
    var cardEndNo: Any? = null

    @SerializedName("CardType")
    @Expose
    var cardType: Any? = null

    @SerializedName("CardIssuer")
    @Expose
    var cardIssuer: Any? = null

    @SerializedName("BankName")
    @Expose
    var bankName: Any? = null

    @SerializedName("PayedByName")
    @Expose
    var payedByName: Any? = null

    @SerializedName("PaymentStatus")
    @Expose
    var paymentStatus: Long = 0

    @SerializedName("CardBasicInfo")
    @Expose
    var cardBasicInfo: Any? = null

    @SerializedName("IsPayLater")
    @Expose
    var isIsPayLater = false
        private set

    @SerializedName("IsPayLaterTicket")
    @Expose
    var isIsPayLaterTicket = false
        private set

    @SerializedName("IsDummyTicket")
    @Expose
    var isIsDummyTicket = false
        private set

    @SerializedName("PaymentSubSectionInfo")
    @Expose
    var paymentSubSectionInfo: Any? = null

    @SerializedName("IsCommonMultiPayment")
    @Expose
    var isIsCommonMultiPayment = false
        private set

    @SerializedName("TransactionTypeId")
    @Expose
    var transactionTypeId: Long = 0

    @SerializedName("olpIdAlias")
    @Expose
    var olpIdAlias: Any? = null

    @SerializedName("IsMilespointAvailable")
    @Expose
    var isIsMilespointAvailable = false
        private set

    @SerializedName("MilesRedeemReference")
    @Expose
    var milesRedeemReference: Any? = null

    @SerializedName("MilesTransReference")
    @Expose
    var milesTransReference: Any? = null

    @SerializedName("MilesRedeemPoints")
    @Expose
    var milesRedeemPoints = 0.0

    @SerializedName("MilesRedeemAmount")
    @Expose
    var milesRedeemAmount = 0.0

    @SerializedName("IsPayWithRedemOnly")
    @Expose
    var isIsPayWithRedemOnly = false
        private set

    fun setIsPaymentSuccess(isPaymentSuccess: Boolean) {
        isIsPaymentSuccess = isPaymentSuccess
    }

    fun setIsPayLater(isPayLater: Boolean) {
        isIsPayLater = isPayLater
    }

    fun setIsPayLaterTicket(isPayLaterTicket: Boolean) {
        isIsPayLaterTicket = isPayLaterTicket
    }

    fun setIsDummyTicket(isDummyTicket: Boolean) {
        isIsDummyTicket = isDummyTicket
    }

    fun setIsCommonMultiPayment(isCommonMultiPayment: Boolean) {
        isIsCommonMultiPayment = isCommonMultiPayment
    }

    fun setIsMilespointAvailable(isMilespointAvailable: Boolean) {
        isIsMilespointAvailable = isMilespointAvailable
    }

    fun setIsPayWithRedemOnly(isPayWithRedemOnly: Boolean) {
        isIsPayWithRedemOnly = isPayWithRedemOnly
    }

    companion object {
        private const val serialVersionUID = 7308898675065546458L
    }
}